src/Mvc/Mvc.ViewFeatures/src/RazorComponents/RazorComponentResult.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures;
+
+/// <summary>
+/// An <see cref="IResult"/> that renders a Razor Component.
+/// </summary>
+public class RazorComponentResult : IResult
```

Comment:
Since we have changed this to only implement `IResult` and not IActionResult, does it run through the MVC result filters?

Reply:
Yes, MVC runs result filters around `IResult` results.